mapie/metrics.py

```diff
@@ -737,3 +737,255 @@ def hsic(
     coef_hsic = np.sqrt(np.matrix.trace(hsic_mat, axis1=1, axis2=2))
 
     return coef_hsic
+
+
+def _picp(
+    y_true: ArrayLike,
+    y_pred_low: ArrayLike,
+    y_pred_up: ArrayLike
+) -> float:
+    """
+    Calculate the Prediction Interval Coverage Probability (PICP).
+
+    The Prediction Interval Coverage Probability (PICP) is a measure used to
+    estimate the fraction of true labels that lie within the prediction
+    intervals.
+
+    [5] Vilde Jensen, Filippo Maria Bianchi, Stian Norman Anfinsen (2022).
+    Ensemble Conformalized Quantile Regression for Probabilistic Time Series
+    Forecasting.
+
+    Parameters
+    ----------
+    y_true : ArrayLike
+        Array of true labels.
+    y_pred_low : ArrayLike
+        Array of lower bounds of prediction intervals.
+    y_pred_up : ArrayLike
+        Array of upper bounds of prediction intervals.
+
+    Returns
+    -------
+    float
+        Prediction Interval Coverage Probability (PICP).
+
+    Notes
+    -----
+    The PICP is calculated as follows:
+    1. Determine the number of true labels that lie within the prediction
+    intervals.
+       This is done by counting the number of elements that satisfy both
+       conditions:
+       (y_true >= y_pred_low) and (y_true <= y_pred_up). Let this count be
+       'in_the_range'.
+    2. Calculate the coverage by dividing 'in_the_range' by the total number
+    of true labels.
+
+    Examples
+    --------
+    >>> y_true = [5, 7.5, 9.5, 10.5, 12.5]
+    >>> y_pred_low = [4, 6, 9, 8.5, 10.5]
+    >>> y_pred_up = [6, 9, 10, 12.5, 12]
+    >>> y_true = np.array(y_true)
+    >>> y_pred_low = np.array(y_pred_low)
+    >>> y_pred_up = np.array(y_pred_up)
+    >>> print(_picp(y_true, y_pred_low, y_pred_up))
+    0.8
+    """
+
+    # Ensure inputs are NumPy arrays for consistent operations
+    y_true = np.asarray(y_true)
+    y_pred_low = np.asarray(y_pred_low)
+    y_pred_up = np.asarray(y_pred_up)
```

Comment:
SUGGESTION: as proposed in previous metrics
```suggestion
    y_true = cast(NDArray, column_or_1d(y_true))
    y_pred_low = cast(NDArray, column_or_1d(y_pred_low))
    y_pred_up = cast(NDArray, column_or_1d(y_pred_up))
```